dashboard/app/coverage.go

```diff
@@ -50,10 +50,22 @@ func GetCoverageDBClient(ctx context.Context) spannerclient.SpannerClient {
 
 type funcStyleBodyJS func(
 	ctx context.Context, client spannerclient.SpannerClient,
-	scope *cover.SelectScope, onlyUnique bool, sss, managers []string,
+	scope *coveragedb.SelectScope, onlyUnique bool, sss, managers []string,
 ) (template.CSS, template.HTML, template.HTML, error)
 
 func handleCoverageHeatmap(c context.Context, w http.ResponseWriter, r *http.Request) error {
+	if r.FormValue("json") == "1" {
+		hdr, err := commonHeader(c, r, w, "")
+		if err != nil {
+			return err
+		}
+		ns := hdr.Namespace
+		repo, _ := getNsConfig(c, ns).mainRepoBranch()
+		// Gzip is the only way to send this data. It is too big.
```

Comment:
I think we might actually benefit from gzipping the responses of many other handlers as well.
What about https://pkg.go.dev/github.com/NYTimes/gziphandler or turning the code you wrote into a more generic handler wrapper?

Reply:
https://github.com/google/syzkaller/issues/5731 to not forget.